### ISSUE TITLE
Why the weight of some task goes to negative?

### ISSUE THREAD

**Issue by nan0755** [ignaciorlando/pytorch-grad-norm#2]
In my experiment, I have 4 sub-tasks for training.  In the first several epcohs, the training procedure worked well, while the weight of some task goes to negative, the whole training procedure may be exploded, and the loss alse goes to negative.
Have you encountered this problem?

Thanks for your reply in advance!

**Comment by farazsaeedan** [comment]
I also have the same probelm where the weight for some tasks goes to negative. I would like to know if you did anything to prevent this.

**Comment by ignaciorlando** [comment]
Hey guys! Sorry I didn't reply before. I've never finished implementing this because I couldn't reproduce the toy example from the paper. I contacted the author but he couldn't help me by that time, so I quited. Perhaps you can contact him to know how to solve this!

**Comment by nan0755** [comment]
I only insert a `ValueError` in script for stop training, and I can not find where goes wrong.

**Comment by nan0755** [comment]
I think the $G^{(i)}_w(t)$ and $r_i(t)$ are calculate as follow:
```python
# G^{(i)}_w(t)
        norms = []
        for j in range(len(self.losses)):
            # get the gradient of this task loss with respect to the shared parameters
            gygw = torch.autograd.grad(self.loss_weights[j]*self.losses[j], W.parameters(),
                                       retain_graph=True, create_graph=True)
            # compute the norm
            norms.append(torch.norm(gygw[0]))  # default norm=2
        norms = torch.stack(norms)
        # compute the inverse training rate r_i(t)
        loss_ratio = torch.Tensor(4).cuda()
        for i in range(len(self.losses)):
            loss_ratio[i] = (self.losses[i] / (self.origin_task_loss[i] + 1e-8))
        # r_i(t)
        inverse_train_rate = (loss_ratio / torch.mean(loss_ratio)).cuda()
```
